core/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -179,11 +180,15 @@ public static CharArraySet parseCommonWords(Environment env, Settings settings,
     }
 
     public static CharArraySet parseArticles(Environment env, Settings settings) {
-        return parseWords(env, settings, "articles", null, null, settings.getAsBoolean("articles_case", false));
+        org.elasticsearch.Version version = settings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, org.elasticsearch.Version.CURRENT);
+        boolean articlesCase = settings.getAsBooleanLenientForPreEs6Indices(version, "articles_case", false);
+        return parseWords(env, settings, "articles", null, null, articlesCase);
     }
 
     public static CharArraySet parseStopWords(Environment env, Settings settings, CharArraySet defaultStopWords) {
-        return parseStopWords(env, settings, defaultStopWords, settings.getAsBoolean("stopwords_case", false));
+        org.elasticsearch.Version version = settings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, org.elasticsearch.Version.CURRENT);
```

Comment:
> If Settings#getAsVersion shouldn't have a default that is another PR anyway.

I felt the same. I will change the call-sites I introduced to throw an exception if no version is defined in the settings.